presto-hive/src/main/java/com/facebook/presto/hive/LocationHandle.java

```diff
@@ -73,14 +83,38 @@ public String getJsonSerializableTargetPath()
     }
 
     @JsonProperty("writePath")
-    public Optional<String> getJsonSerializableWritePath()
+    public String getJsonSerializableWritePath()
     {
-        return writePath.map(Path::toString);
+        return writePath.toString();
     }
 
     @JsonProperty("isExistingTable")
     public boolean getJsonSerializableIsExistingTable()
     {
         return isExistingTable;
     }
+
+    @JsonProperty("writeMode")
+    public WriteMode getJsonSerializableWriteMode()
+    {
+        return writeMode;
+    }
+
+    public enum WriteMode
+    {
+        STAGE_AND_MOVE_TO_TARGET_DIRECTORY, // common mode for new table or existing table (both new and existing partition)
```

Comment:
Make these all Javadoc comments, so that someone looking at Javadoc for the constants will see it.